ocs_ci/deployment/baremetal.py

```diff
@@ -537,6 +537,39 @@ def create_pxe_files(self, ocp_version, role):
                 t_file.writelines(default_pxe_file)
             return temp_file.name
 
+        def set_pxe_boot_and_reboot(self, machine):
+            """
+            Ipmi Set Pxe boot and Restart the machine
+
+            Args:
+                machine (str): Machine Name
+
+            Returns:
+
+            """
+            secrets = [
+                self.mgmt_details[machine]["mgmt_username"],
+                self.mgmt_details[machine]["mgmt_password"],
+            ]
+            # Changes boot prioriy to pxe
+            cmd = (
+                f"ipmitool -I lanplus -U {self.mgmt_details[machine]['mgmt_username']} "
+                f"-P {self.mgmt_details[machine]['mgmt_password']} "
+                f"-H {self.mgmt_details[machine]['mgmt_console']} chassis bootdev pxe"
+            )
+            run_cmd(cmd=cmd, secrets=secrets)
+            sleep(2)
```

Comment:
We should log why we are sleeping here

Reply:
sleep was added because ipmi cmd get execute within sec so sometimes bootdev pxe is not set that's the reason sleep was added

Reply:
That makes sense. We should log that we are sleeping for that reason.

Reply:
Done